model/vm_host.rb

```diff
@@ -243,16 +244,83 @@ def reset
     Hosting::Apis.reset_server(self)
   end
 
+  def check_storage_smartctl(ssh_session, devices)
+    devices.map do |device_name|
+      result = ssh_session.exec!("sudo smartctl -j -H /dev/#{device_name} | jq .smart_status.passed").strip == "true"
+      result.tap do |passed|
+        Clog.emit("Device #{device_name} failed smartctl check.") unless passed
+      end
+    end.all?(true)
+  end
+
+  def check_storage_nvme(ssh_session, devices)
+    devices.map do |device_name|
+      result = ssh_session.exec!("sudo nvme smart-log /dev/#{device_name} | grep \"critical_warning\" | awk '{print $3}'").strip == "0"
+      result.tap do |passed|
+        Clog.emit("Device #{device_name} failed nvme smart-log check.") unless passed
+      end
+    end.all?(true)
+  end
+
+  def check_storage_read_write(ssh_session, devices)
+    lsblk_json_info = ssh_session.exec!("lsblk --json")
+    devices_with_mount_points = devices.map { |device| SystemParser.get_device_mount_points_from_lsblk_json(lsblk_json_info, device) }
+
+    all_mount_points = []
+    devices_with_mount_points.each do |device_mount_points|
+      device_mount_points.each do |device_name, mount_points|
+        mount_points.each do |mount_point|
+          all_mount_points.append(mount_point)
+        end
```

Comment:
```suggestion
        all_mount_points.concat(mount_points)
```

Reply:
What do you think of using `flat_map`s to do the job?

Reply:
That approach is fine, but it likely results in more intermediate arrays, and with nested `flat_map`s, I think it's a little less clear.